v2/robotmk/scheduler.py

```diff
@@ -173,9 +177,9 @@ def __call__(self) -> None:
             return  # Untested
 
         final_output = retry_spec.output_directory() / "merged.xml"
-
         rebot(*outputs, output=final_output, report=None, log=None)
-        self._final_outputs.append(final_output)
+
+        self._write_result_file(final_output)
```

Comment:
OTI: This function now has multiple side-effects. I think we should maybe review this with Mathias?

Reply:
I would say that after this commit, the scheduler is in a minimum viable product stage. I would suggest to do two steps:

1. We sit together internally and discuss the code quality. We can also do immediate improvements such as renamings.
2. We discuss/review our code with eg. Mathias.

What do you think about this?